modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/property/PropertiesHolderImpl.java

```diff
@@ -38,38 +40,45 @@ class PropertiesHolderImpl implements PropertiesHolder {
     }
 
     /** {@inheritDoc} */
-    @SuppressWarnings("unchecked")
     @Override
-    public <T> @Nullable T get(Property<T> prop) {
-        Object val = props.get(prop);
+    public <T> T get(Property<T> prop) {
+        T val = get0(prop);
 
         if (val == null) {
-            return null;
+            throw new RuntimeException(format("Property '{}' is not set", prop.name));
```

Comment:
seems right exception is PropertyNotFound

Reply:
good catch! Fixed